resilience4j-circuitbreaker/src/main/java/io/github/resilience4j/circuitbreaker/internal/CircuitBreakerStateMachine.java

```diff
@@ -154,6 +156,14 @@ public String toString() {
         return String.format("CircuitBreaker '%s'", this.name);
     }
 
+    @Override
+    public void resetToClosedState() {
+        CircuitBreakerState previousState = stateReference.getAndUpdate(currentState -> new ClosedState(this));
+        if (previousState.getState() != CLOSED) {
```

Comment:
We should publish a StateResetEvent, even if the state is already closed, because the metrics have been reset. It's possible to do a state reset and state transition from closed to closed now.
That means we also need a new StateTransition enum `FORCED_CLOSED_TO_CLOSED` and we have to adapt the method  `transitionToClosedState` and add a new switch case.
Correct?

Reply:
Well resetToClosed is an idempotent operation, I would simply publish state changes if they happen, and publish reset in any case. Do you prefer me to publish one or two events when resetting? If it wasn't CLOSED -> StateTransition and StateReset (maybe then would be changing to RESET), if CLOSED -> RESET

Reply:
Oh I see, I missed the if in the refactoring.
But yes, the more I think about it, the more I think I should publish two events, and not have RESET a copy of STATE_TRANSITION

Reply:
That's fine for me. We can publish two events. One for state transition and one for state reset.
Then we don't have to add a new enum to StateTransition and we don't have to publish an event when reset is triggered in the closed state.